src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -286,11 +286,13 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite with SQLHelper {
       "'appendedFiles' are not usable indexes if hybrid scan is disabled.") {
     withSQLConf(INDEX_HYBRID_SCAN_ENABLED -> "false") {
       val entry1 = createIndexLogEntry("t1iTest", Seq(t1c1), Seq(t1c3), t1ProjectNode)
-      val entry2 = entry1.withAppendedAndDeletedFiles(Seq(), Seq("f1"))
-      val entry3 = entry1.withAppendedAndDeletedFiles(Seq("f2"), Seq())
+      val entry2 = entry1.withAppendedAndDeletedFiles(Seq(), Seq(FileInfo("file:/dir/f1", 1, 1)))
+      val entry3 = entry1.withAppendedAndDeletedFiles(Seq(FileInfo("file:/dir/f2", 1, 1)), Seq())
+      entry2.state = "ACTIVE"
+      entry3.state = "ACTIVE"
```

Comment:
Is this change required? (It helps reviewers if you make guide comments on changes that are not PR-related).

Reply:
Oh I see. This is very subtle and I will create an issue to see if we need to follow up (potentially any `copy()` on `IndexLogEntry` could be misleading).

Meanwhile could you add the following?
```
// IndexLogEntry.withAppendedAndDeletedFiles doesn't copy LogEntry's fields. Thus, set the 'state' to ACTIVE manually so that these entries are considered in RuleUtils.getCandidateIndexes.
```